### PR TITLE
man: allow command to run on unsupported platforms when --force is passed

### PR DESCRIPTION
**Pull request by dtrodrigues**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
- [x] Have you successfully run `brew man` locally and committed any changes?

-----
The `brew man` command doesn't work on Apple silicon using the system ruby. However, building and using https://github.com/Homebrew/homebrew-portable-ruby does allow the `ronn` gem and therefore the `brew man` command to work. Given that this is not a supported configuration, allow the command to run on Silicon only if a user passes the `--force` flag.

Helps to address https://github.com/Homebrew/brew/issues/10210

### CODE REVIEW

**Review by MikeMcQuaid — APPROVED** [review]
I'd prefer we didn't do this. We don't ship this portable ruby so it's not a supported workflow. If we really need maintainers to be able to do this: let's make it use an undocumented environment variable.

**Comment by dtrodrigues** [comment]
I don't like needing this either. Changing to a developer environment variable does seem preferable, so I updated it to use the `HOMEBREW_SILICON_DEVELOPER` variable.

I'll open a separate issue describing the current state of the system ruby on Apple Silicon to discuss potential paths forward for contributors on Apple Silicon.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Looks good, thanks @dtrodrigues!

**Pull request merged**
